solution.cpp

```diff
@@ -23,112 +23,192 @@ using std::ifstream;
- * `outfilename`, separated by newlines.
- *
- * Arguments:
- * - `size`: The size of (the number of elements in) the array.
- * - `primes`: An array of boolean values, with each element set to `true` if
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_composites( const int size,
-                      bool primes[],
-                      const string& infilename,
-                      const string& outfilename );
+* Read numbers from `infilename`, and if they are composite, output them to
+* `outfilename`, separated by newlines.
+*
+* Arguments:
+* - `size`: The size of (the number of elements in) the array.
+* - `primes`: An array of boolean values, with each element set to `true` if
+*   its index is prime, and `false` otherwise.
+* - `infilename`: A string representing the file path to read from.
+* - `outfilename`: A string representing the file path to write to.
+*
+* Returns:
+* - 0 on success, or -1 on error.
+*/
+int write_composites(const int size,
+	bool primes[],
+	const string& infilename,
+	const string& outfilename);
 
 // ----------------------------------------------------------------------------
 
 int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
-    }
-
-    bool primes[max+1];
-    sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
-    }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
-    }
-
-    return 0;  // success
+	int max = find_max("input.txt");
+	cout << max;
+	if (max == -1) {
+		cout << "ERROR in `find_max()`" << endl;
+		return 1;  // error
+	}
+
+	bool * primes = (bool *)malloc(sizeof(bool) * (max + 1));;
+	sieve(max + 1, primes);
+
+	int ret;  // for storing return values, to check for error codes
+	ret = write_primes(max + 1, primes, "input.txt", "primes.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_primes()`" << endl;
+		return 1;  // error
+	}
+	//
+	ret = write_composites(max + 1, primes, "input.txt", "composites.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_composites()`" << endl;
+		return 1;  // error
+	}
+	system("pause");
+	return 0;  // success
 }
 
 // ----------------------------------------------------------------------------
 
 // TODO: write prototyped functions
 
+	int find_max(const string& infilename) {
+	int a = 0; //declares a return variable
```

Comment:
better to use more descriptive names, like `max` and `current`, rather than `a` and `b`, or something similar